### DISCUSSION TITLE
Is there anything to learn from "Draco"/can we learn from "Draco"?

### DISCUSSION THREAD

**Discussion by balacij** [JacquesCarette/Drasil#3025] [Ideas]
I'm re-reading the ["Overview of Generative Software Development" (Czarnecki2004)](https://gitlab.cas.mcmaster.ca/smiths/pub/-/blob/master/Czarnecki2004.pdf), and this pass is a lot more informative after working with Drasil for a while (and through exploring Drasil in #2883). In Chapter 4.2 of the paper, Czarnecki referenced "Jim Neighbors"' [phd thesis](http://bayfronttechnologies.com/thesis.pdf), focusing on ["Draco"](https://www-di.inf.puc-rio.br/~julio/Slct-pub/draco.pdf), which, at face value, appears to be very similar to Drasil ideologically. 

I've only very quickly skimmed through the second linked right now, but I figured I should share it here. Its interesting to see that some of the terminology (e.g., refinements) is naturally similar to Drasils.

Figure 27 of Jim Neighbors PhD thesis is a neat example to quickly look at, if you want to get a quick idea.

(Sorry, I don't know if this should have been an email to all of us because it doesn't quite fit any of our discussion categories)

**Comment by balacij** [comment]
There are also some [related resources](https://bayfronttechnologies.com/l01tech.htm) on the Jim Neighbors' [company website](https://bayfronttechnologies.com/).

**Reply by balacij** [reply]
[Another page](https://bayfronttechnologies.com/l02draco.htm) with many listed papers.

**Comment by smiths** [comment]
Thank you @balacij for the discussion topic.  This seems like the appropriate spot for this kind of discussion to me.  If you sent it over e-mail then eventually we might forget that any conversation took place.

I agree that the Draco example (as shown in Figure 27) has some elements that are reminiscent of Drasil.  I especially like that they have citations to where the original information/algorithms came from.

I wonder if this would be a good thing to discuss during one of our Drasil meetings?  That reminds me that I have to create the agenda for this Friday's meeting.  :smile:

**Reply by balacij** [reply]
> I wonder if this would be a good thing to discuss during one of our Drasil meetings?

It might just be! I'm going to try to look into it a bit more later, and see what I can find.

**Comment by balacij** [comment]
@JacquesCarette [looked into this](https://github.com/JacquesCarette/Drasil/wiki/Related-Work#draco).
